django_airavata/apps/api/views.py

```diff
@@ -184,6 +184,12 @@ def perform_create(self, serializer):
         experiment = serializer.save(
             gatewayId=self.gateway_id,
             userName=self.username)
+        #FIXME: Moving the setting experimentDataDirectory on experiment step during experiment creation. Check with the team
+        self._set_storage_id_and_data_dir(experiment)
```

Comment:
This should mostly be okay. I implemented it so that the experiment data directory is created at launch time because the user can change the experiment's name and project after creating the experiment and the experiment name and project name are used in generating the experiment data directory. So my thinking was might as well wait until the last moment to create it. But this is fine too.

However, since the experiment data directory isn't really needed (it's just used to get the user's directory in the storage resource), maybe there's another way. I thought why not use the StoragePreference.fileSystemRootLocation + the user's username to get the user's directory. Did you consider that approach?

Reply:
Hi @machristie ,

Thanks for reviewing the code.
I did not consider this approach, thanks for letting me know. But if I need to go with the current approach, the '_set_storage_id_and_data_dir()' API needs to be called (Please check my latest comments on my Airavata PR). I've posted most of my findings over there. Please let me know if it makes sense to stick with the current approach or if I should go with my initial approach.

Reply:
I think if you integrate the check with launchExperiment instead of createExperiment (which I just now suggested on the other PR), then you don't need `_set_storage_id_and_data_dir` in `perform_create`.